homeassistant/components/onewire/onewirehub.py

```diff
@@ -54,22 +56,23 @@ async def initialize(self, config_entry: ConfigEntry) -> None:
             await self.connect(host, port)
         await self.discover_devices()
 
-    async def discover_devices(self):
+    async def discover_devices(self) -> None:
         """Discover all devices."""
         if self.devices is None:
             if self.type == CONF_TYPE_SYSBUS:
+                assert self.pi1proxy
                 self.devices = await self.hass.async_add_executor_job(
                     self.pi1proxy.find_all_sensors
                 )
             if self.type == CONF_TYPE_OWSERVER:
                 self.devices = await self.hass.async_add_executor_job(
                     self._discover_devices_owserver
                 )
-        return self.devices
 
-    def _discover_devices_owserver(self, path="/"):
+    def _discover_devices_owserver(self, path: str = "/") -> list:
```

Comment:
```suggestion
    def _discover_devices_owserver(self, path: str = "/") -> list[dict[str, str]]:
```

Reply:
I created a `OWServerDeviceDescription` TypedDict for the OWServer device listing